assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.js

```diff
@@ -21,15 +21,97 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+import { MetricTileTable } from '../../../../components/KeyMetrics';
+const { useSelect, useInViewSelect } = Data;
+
 export default function TopCitiesWidget( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const topcCitiesReportOptions = {
+		...dates,
+		dimensions: [ 'city' ],
+		metrics: [ { name: 'totalUsers' } ],
+		orderby: [
+			{
+				metric: {
+					metricName: 'totalUsers',
+				},
+				desc: true,
+			},
+		],
+		limit: 3,
+	};
+
+	const topCitiesReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( topcCitiesReportOptions )
+	);
+
+	const loading = useInViewSelect(
```

Comment:
The in-view select variant is only intended for selectors that get reports since it will return `undefined` unconditionally when not in view which would not be expected for a selector like `hasFinishedResolution` which would never return anything other than a boolean. It could be used for other things too but it's essentially intended for guarding selectors that have resolvers and their respective side-effects.

Reply:
Thanks, I have updated it to use `useSelect`. However, I have checked other KM widgets, and they use the `useInViewSelect` selector, which needs to be addressed.